Add lutron caseta model Caseta Shade SerenaEssentialsRollerShade

TiddlyWiddly opened the pull request:
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Add support for new model roller shade SerenaEssentialsRollerShade, SYERX-B-X

This should fix my bug/feature request: https://github.com/home-assistant/core/issues/139614


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes https://github.com/home-assistant/core/issues/139614
- This PR is related to issue: 
- Link to documentation pull request: 
- Link to developer documentation pull request: 
- Link to frontend pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

TiddlyWiddly committed:
Update cover.py

Add support for new model roller shade SerenaEssentialsRollerShade, SYERX-B-X

TiddlyWiddly committed:
Merge branch 'dev' into patch-1

MartinHjelmare commented:
Have you tested the change?

MartinHjelmare reviewed (dismissed):
Please solve the CI failure.

eclair4151 commented:
You must add support for the new device type to the pylutron caseta library first, then raise a PR to bump the library version, then you can raise the PR here to reference the new device type. 

See https://github.com/gurumitts/pylutron-caseta/pull/178

eclair4151 reviewed (dismissed):
add support in pylutron caseta and bump pip version in dependency list first

TiddlyWiddly committed:
update requirements

	modified:   homeassistant/components/lutron_caseta/cover.py
	modified:   homeassistant/components/lutron_caseta/manifest.json
	modified:   requirements_all.txt
	modified:   requirements_test_all.txt

TiddlyWiddly commented:
@eclair4151 merged upstream library changes, tested in library, bumped version, bumped requirements, and tested locally in HA.

https://github.com/gurumitts/pylutron-caseta/pull/184

bdraco commented on homeassistant/components/lutron_caseta/manifest.json:
```diff
@@ -9,7 +9,7 @@
   "iot_class": "local_push",
   "loggers": ["pylutron_caseta"],
-  "requirements": ["pylutron-caseta==0.23.0"],
+  "requirements": ["pylutron-caseta==0.24.0"],
```
Please bump the library in a separate PR

We should only have one box the `Type of change` boxes checked per PR.

Thanks!

bdraco reviewed (dismissed):
Please see https://github.com/home-assistant/core/pull/139800/files#r1988389807

bdraco commented:
I took care of opening a PR for the bump in https://github.com/home-assistant/core/pull/140338. Once that merges, please rebase this PR

bdraco committed:
Merge branch 'dev' into patch-1

TiddlyWiddly commented:
@bdraco thank you for splitting and opening the other PR. do I need to do anything else on this or are just waiting for review and merge?

bdraco commented:
> @bdraco thank you for splitting and opening the other PR. do I need to do anything else on this or are just waiting for review and merge?

Nothing else to do here. I needed to re-check all the review comments to make sure they had been addressed and clear the review state before merging. That's done now

bdraco commented:
Thanks @TiddlyWiddly
